pipeline/frontend/yaml/compiler/option.go

```diff
@@ -226,6 +230,13 @@ type ProxyOptions struct {
 // and NO_PROXY environment variables added by default to every
 // container in the pipeline.
 func WithProxy(opt ProxyOptions) Option {
+	// if options are all empty we don't set any env var
+	if opt.HTTPProxy == "" &&
+		opt.HTTPSProxy == "" &&
+		opt.NoProxy == "" {
+		return noopOption()
+	}
+	// else we pass on as upper- and lower-case
```

Comment:
What do you mean by upper- and lower-case? I would just leave the comment away is quite self explaining already.